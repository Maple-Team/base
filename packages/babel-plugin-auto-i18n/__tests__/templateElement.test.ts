import { getTransformCode } from './test-help'

describe('arrow function declaration scenarios - templateElement cases', () => {
  describe('exported arrow function cases', () => {
    it('handle templateElement case 1', () => {
      const sourceCode = `
        export const Component = () => {
          const var1 = \`我好\`
          return <div>{\`你好\`}</div>
        }`
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle templateElement case 2', () => {
      const sourceCode = `
        export const Component = () => {
          const var1 = \`我好\`
          console.log(\`不好\`)
          return <div>{\`你好\`}</div>
        }`
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle templateElement case 3', () => {
      const sourceCode = `
        export const Component = () => {
          const var1 = \`我好\`
          console.log(\`不好\`)
          console.log({
            a: \`不好\`
          })
          return <div>{\`你好\`}</div>
        }`
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle templateElement wrapper with jsxFragment case 1', () => {
      const sourceCode = `
        export const Component = () => {
          const var1 = \`我好\`
          return <><div>{\`你好\`}</div></>
        }`
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle templateElement wrapper with jsxFragment case 2', () => {
      const sourceCode = `
        export const Component = () => {
          const var1 = \`我好\`
          return <></>
        }`
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })
})
