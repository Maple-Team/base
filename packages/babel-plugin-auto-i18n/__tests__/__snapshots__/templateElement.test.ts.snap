// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrow function declaration scenarios - templateElement cases exported arrow function cases handle templateElement case 1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var var1 = t('key_我好');
  return <div data-i18n=\\"[\\\\\\"key_\\\\u6211\\\\u597D\\\\\\",\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t('key_你好')}</div>;
};"
`;

exports[`arrow function declaration scenarios - templateElement cases exported arrow function cases handle templateElement case 2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var var1 = t('key_我好');
  console.log(\\"\\\\u4E0D\\\\u597D\\");
  return <div data-i18n=\\"[\\\\\\"key_\\\\u6211\\\\u597D\\\\\\",\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t('key_你好')}</div>;
};"
`;

exports[`arrow function declaration scenarios - templateElement cases exported arrow function cases handle templateElement case 3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var var1 = t('key_我好');
  console.log(\\"\\\\u4E0D\\\\u597D\\");
  console.log({
    a: \\"\\\\u4E0D\\\\u597D\\"
  });
  return <div data-i18n=\\"[\\\\\\"key_\\\\u6211\\\\u597D\\\\\\",\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t('key_你好')}</div>;
};"
`;

exports[`arrow function declaration scenarios - templateElement cases exported arrow function cases handle templateElement wrapper with jsxFragment case 1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var var1 = t('key_我好');
  return <><div data-i18n=\\"[\\\\\\"key_\\\\u6211\\\\u597D\\\\\\",\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t('key_你好')}</div></>;
};"
`;

exports[`arrow function declaration scenarios - templateElement cases exported arrow function cases handle templateElement wrapper with jsxFragment case 2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var var1 = t('key_我好');
  return <><div data-i18n='[\\"我好\\"]'></div></>;
};"
`;
